lib/elements/rh-context-picker/rh-context-picker.ts

```diff
@@ -1,58 +1,113 @@
 
   static readonly styles = [style];
 
-  static readonly palettes: ColorPalette[] = [
-    'darkest',
-    'darker',
-    'dark',
-    'light',
-    'lighter',
-    'lightest',
-  ];
+  static readonly palettes = new Map<ColorPalette, Color>(Object.entries({
+    darkest,
+    darker,
+    dark,
+    light,
+    lighter,
+    lightest,
+  }) as [ColorPalette, Color][]);
+
+  private static offsets: Partial<Record<ColorPalette, number>> = {
+    darkest: -4,
+    darker: -3,
+    dark: -3,
+    light: -1,
+    lighter: 1,
+    lightest: 2,
+  };
+
+  private static paletteNames = Array.from(RhContextPicker.palettes, ([name]) => name);
 
   declare shadowRoot: ShadowRoot;
 
-  #internals = this.attachInternals();
-
-  #target: HTMLElement | null = null;
-
   /** ID of context element to toggle (same root) */
   @property() target?: string;
 
-  @property() value?: ColorPalette;
+  @property() value: ColorPalette = 'darkest';
+
+  @query('#context-range') private range?: HTMLInputElement;
+
+  @colorContextConsumer() private on?: ColorTheme;
 
-  @query('#context-range') range?: HTMLInputElement;
+  #offset = RhContextPicker.offsets[this.value];
+
+  #internals = this.attachInternals();
+
+  #target: HTMLElement | null = null;
+
+  willUpdate() {
+    this.#offset = RhContextPicker.offsets[this.value];
+  }
 
   render() {
+    const { on = 'dark', value = 'darkest' } = this;
+    const derivedLabel = this.#internals.ariaLabel ?? Array.from(this.#internals.labels, x => x.textContent).join();
     return html`
-      <label for="context-range">Color Palette</label>
-      <input id="context-range"
-             name="range"
-             type="range"
-             list="palettes"
-             max="5"
-             @input="${this.#onInput}">
-      <datalist id="palettes">
-        <option value="0" label="darkest"></option>
-        <option value="1" label="darker"></option>
-        <option value="2" label="dark"></option>
-        <option value="3" label="light"></option>
-        <option value="4" label="lighter"></option>
-        <option value="5" label="lightest"></option>
-      </datalist>
+      <div id="container" class="${classMap({ [on]: true })}">
+          <input id="context-range"
+                 class="${classMap({ [value]: true })}"
+                 name="range"
+                 type="range"
+                 list="palettes"
+                 max="5"
+                 aria-label="${derivedLabel}"
+                 style="${styleMap({ '--offset': `${this.#offset}px` })}"
+                 @input="${this.#onInput}">
+          <datalist id="palettes">${Array.from(RhContextPicker.palettes, ([palette]) => html`
+            <option id="option-${palette}"
+                    value="${palette}"
+                    title="${palette}"
+                    @click="${() => this.#setValue(palette)}">
+              <span class="visually-hidden">${palette}</span>
+            </option>`)}
+          </datalist>
```

Comment:
Thinking this should be a radio group instead so we can utilize `:checked` for current active color palette and highlight that visually.

Reply:
sounds reasonable to me. @nikkimk wdyt?

Reply:
could be radio group or a select

Reply:
let's do this in a follow up PR, k?